### PR TITLE
Automated deployment to PyPI on release

### PR DESCRIPTION
**Pull request by ianmkenney**
The package should automatically be uploaded to PyPI when a release is made.

### CODE REVIEW

**Comment by ianmkenney** [comment]
@IAlibay Could you take a look at the `deploy.yaml` workflow? I took it from `mdaencore`, which you made. If this does the job, then it should be straightforward to move to other MDAKits.

**Comment by IAlibay** [comment]
> @IAlibay Could you take a look at the `deploy.yaml` workflow? I took it from `mdaencore`, which you made. If this does the job, then it should be straightforward to move to other MDAKits.

By other MDAKit you mean "other Cython using kits" right? i.e. cibuildwheels isn't suitable for pure python kits

**Review by IAlibay — CHANGES_REQUESTED** [review]
For pure python you should instead use the pypi-deployment action (I assume PSA is indeed pure python).

**Comment by ianmkenney** [comment]
I figured it wouldn't be the best reference for that reason, I'll just rewrite from scratch then! Thanks

**Comment by IAlibay** [comment]
> I figured it wouldn't be the best reference for that reason, I'll just rewrite from scratch then! Thanks

Please don't rewrite from scratch and use the reusable action - it is there so we can align across all our repos

File: .github/workflows/deploy.yaml
Comment (IAlibay) [inline .github/workflows/deploy.yaml:1]: Essentially you want to just have this workflow file: https://github.com/MDAnalysis/mda-xdrlib/blob/main/.github/workflows/deploy.yaml

**Review by orbeckst — CHANGES_REQUESTED** [review]
I assume we can't test if it works without merging? So we'll merge this and then do a 1.0.1 release?

The PyPi metadata doesn't have a link to a documentation page. Enable https://github.com/MDAnalysis/PathSimAnalysis/blob/3267deeb0f436732b69783d88a49ba7a34f07687/pyproject.toml#L42-L44 for docs and source.

MDAnalysis is not the only dependency in https://github.com/MDAnalysis/PathSimAnalysis/blob/3267deeb0f436732b69783d88a49ba7a34f07687/pyproject.toml#L21 — we also have
```python
import numpy as np
from scipy import spatial, cluster
from scipy.spatial.distance import directed_hausdorff
import matplotlib
```
so they all need to be explicitly listed in `pyproject.toml` (important for PyPi!) and also in [devtools/conda-envs/test_env.yaml](https://github.com/MDAnalysis/PathSimAnalysis/blob/main/devtools/conda-envs/test_env.yaml).

**Review by orbeckst — APPROVED** [review]
This _looks_ ok to me. Let's first see if https://github.com/MDAnalysis/waterdynamics/pull/8 works, given that it's essentially implementing the same workflow. If any problems emerge there then we can fix them here, too.

Assuming all is fine, we can then do a 1.0.1 here.

**Pull request merged**